### PR TITLE
Chore/test/course component

### PR DESCRIPTION
**Pull request by nicolasruscher**
### Checklist
- [X] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [X] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [X] Client: I documented the TypeScript code using JSDoc style.

### Motivation and Context
Added tests to client components to increase overall test coverage 

### Description

- 

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. ...

### Test Coverage

- course-overview-component: 84%
- course-component: 90.91%

### CODE REVIEW

**Review by SimonEntholzer — APPROVED** [review]
The ode looks good :D

File: src/test/javascript/spec/component/course/course-overview.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        component.ngOnInit();
+        tick(1000);
+
+        expect(getCourseStub).to.have.been.called;
+        expect(adjustCourseDescriptionStub).to.have.been.called;
+        expect(subscribeForQuizChangesStub).to.have.been.called;
+        expect(subscribeToTeamAssignmentUpdatesStub).to.have.been.called;
+    }));
+
+    it('Should call load Course methods on init', fakeAsync(() => {
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/course/course-overview.component.spec.ts:137]: I don't understand this test case, isn't it exactly the same like the one above just without `getCourseStub.returns(course1);`? (Which we don't test for anyway?)

File: src/test/javascript/spec/component/course/course-overview.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        expect(subscribeForQuizChangesStub).to.have.been.called;
+        expect(subscribeToTeamAssignmentUpdatesStub).to.have.been.called;
+    }));
+
+    it('should set Long Description', () => {
+        component.longTextShown = false;
+
+        component.showLongDescription();
+
+        expect(component.courseDescription).to.equal('');
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/course/course-overview.component.spec.ts:159]: I'd use `.to.be.empty`.

File: src/test/javascript/spec/component/course/course-overview.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+        expect(component.longTextShown).to.equal(true);
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/course/course-overview.component.spec.ts:160]: We usually use `.to.be.true` (Same with a few other `.equals(false/true)` below.)

**Review by FrankeLukas — APPROVED** [review]
Code changes generally lgtm. A few questions/remarks and inline comments:
- You use `sinon.spy` etc. already, so we probably want to use `sinon.stub` too?
- Does using `fixture.detectChanges` not work in the two tests you faked async?

**Review by krusche — APPROVED** [review]

**Pull request merged**